src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs

```diff
@@ -249,12 +249,16 @@ internal decimal Decimal
                                 }
                                 catch (OverflowException)
                                 {
-                                    throw new OverflowException(SQLResource.ConversionOverflowMessage);
+                                    throw new OverflowException(SQLResource.ConversionOverflowMessage + " for " + sqlValue.ToString());
                                 }
                             }
                         }
-                        throw new OverflowException(SQLResource.ConversionOverflowMessage);
+                        throw new OverflowException(SQLResource.ConversionOverflowMessage+ " for " + sqlValue.ToString());
```

Comment:
The sqlValue should be passed as a parameter to a function which uses a fomatter to correctly display the message in the user's langauge. Have a look at other similar exception messages that have parameter data in the project.